scipy/signal/ltisys.py

```diff
@@ -2494,3 +2497,279 @@ def place_poles(A, B, poles, method="YT", rtol=1e-3, maxiter=30):
+                return :math:`\bar C = C P` otherwise `Cbar = None` will be
+                returned. Cbar has the following form:
+
+                 .. math::
+                     \bar C = \left[ \begin{array}{cc}
+                                     C_c & C_{nc}
+                                     \end{array} \right]
+
+                With :math:`C_c` such that the transfer function of (A,B,C) and
+                (Hc,Bc,Cc) are the same.
+
+            P : ndarray
+                The orthogonnal similarity transformation
+
+            control_index : int
+                The controllability index, i.e the number of controllable
+                states, of the linear system :math:`AX+BU`. If
+                ``control_index == A.shape[0]`` then the linear system is
+                controllable.
+
+    Examples
+    --------
+    A simple example demonstrating how to check the control index of a linear
+    system.
+
+    >>> from scipy import signal
+
+    >>> A = np.array([[ 1.380,  -0.2077,  6.715, -5.676  ],
+    ...               [-0.5814, -4.290,   0,      0.6750 ],
+    ...               [ 1.067,   4.273,  -6.654,  5.893  ],
+    ...               [ 0.0480,  4.273,   1.343, -2.104  ]])
+    >>> B = np.array([[ 0,      5.679 ],
+    ...               [ 1.136,  1.136 ],
+    ...               [ 0,      0,    ],
+    ...               [-3.146,  0     ]])
+    >>> hessenberg_form = signal.controller_hessenberg(A, B)
+    >>> hessenberg_form.control_index
+    4
+
+    References
+    ----------
+    .. [1] Datta Biswa, "Numerical Methods for Linear Control Systems",
+           Elsevier, pp 173-183, 2003.
+
+    """
+    A = np.asarray(A)
+    B = np.asarray(B)
+    if A.ndim != 2:
+        raise ValueError("A must be a 2D array/matrix.")
+    if B.ndim != 2:
+        raise ValueError("B must be a 2D array/matrix")
+    if C is not None:
+        C = np.asarray(C)
+        if C.ndim != 2:
+            raise ValueError("C must be a 2D array/matrix")
+    if A.shape[0] != A.shape[1]:
+        raise ValueError("A must be square")
+
+    stop = False
+    P = None
+    # H_sub => the submatrix of H we are working on
+    H_sub = B
+
+    # where to seek the submatrices H_sub
+    H_start = 0
+    control_index = 0
+    while not stop:
+        # page 174 step 0
+        P_update, R, _ = s_qr(H_sub, pivoting=True)
+        # be consistent with [1] they use the form Q A P = R
+        # whereas scipy uses A P = Q R. The variables names
+        # are chosen to match [1] therefore P_update refers to "Q" from
+        # the QR decomposition and not the permutation matrix usually
+        # named "P".
+        P_update = P_update.T
+
+        H_start = control_index
+
+        # find the first occurence of a zero line in R, this will
+        # reveal the rank of H_sub
+        zero_lines_idx = np.where(np.all(
+                            np.isclose(R, [0]*R.shape[1]), axis=1))[0]
```

Comment:
I don't think you actually need to do `[0] * R.shape[1]`, doing `np.isclose(R, 0)` should broadcast